### ISSUE TITLE
Component: FieldSet (Screen reader issues)

### ISSUE THREAD

**Issue by olemartinorg** [Altinn/altinn-design-system#248]
### Description of the bug

In https://github.com/Altinn/app-frontend-react/pull/767 we got an external contribution to fix some WCAG issues in the app-frontend List component. The FieldSet from Material UI was used there instead of the one in our design system, apparently because of a lack of proper screen reader support. See the comment in that PR for more details.

### Steps To Reproduce

1. Use FieldSet
2. Start screen reader?

**Comment by olemartinorg** [comment]
Beware: Because the `FieldSet` was not working correctly with a screen reader, the Material UI FieldSet was used instead in the List component in `app-frontend-react`. Please make sure to create an issue in app-frontend-react to resolve that whenever this gets fixed in the design system.

**Comment by Magnusrm** [comment]
Issue to replace MUI FieldSet in app-frontend-react: https://github.com/Altinn/app-frontend-react/issues/902
